### PR TITLE
Adds JSXFragment to plugin-transform-typescript check for the presence of jsx

### LINKED ISSUE #7979: TypeScript JSX Fragment removes React import
## Bug Report

**Current Behavior**
Currently whenever I'm using "@babel/preset-typescript" with JSX fragments and no other JSX, the React import is removed from the compiled output. This only happens with JSX fragments. If you use `React.Fragment` instead, the compiled output is correct. Also worth noting that any other non-fragment jsx in the source will cause the React import to be maintained.

**Input Code**

```tsx
import React, { SFC } from "react";

export const BreadcrumbItem: SFC = ({ children }) => <>{children}</>;
```

**Expected code**

```js
import React from "react";
export var BreadcrumbItem = function BreadcrumbItem(_ref) {
  var children = _ref.children;
  return React.createElement(React.Fragment, null, children);
};
```

**Actual code**

```js
export var BreadcrumbItem = function BreadcrumbItem(_ref) {
  var children = _ref.children;
  return React.createElement(React.Fragment, null, children);
};
```

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
  "presets": [
    "@babel/preset-typescript",
    "@babel/preset-react",
    [
      "@babel/preset-env",
      {
        "modules": false,
        "targets": {
          "browsers": "ie 11"
        },
        "useBuiltIns": "entry"
      }
    ]
  ],
  "plugins": [
    ["emotion", { "sourceMap": true, "autoLabel": true }],
    "@babel/proposal-class-properties",
    "@babel/proposal-object-rest-spread"
  ]
}
```

**Environment**
- Babel version(s): v^7.0.0-beta.47
- Node/npm version: v8.9.4
- OS: Windows 10
- Monorepo no
- How you are using Babel: cli & loader

**Additional context/Screenshots**

I have determined this issue to only happen when when "@babel/preset-typescript" is included. Removing any number of other plugins and presets does not fix issue. Additionally I've tried the following variations of the source and they all had the same incorrect output.

```tsx
import React from "react";
import { SFC } from "react";

export const BreadcrumbItem: SFC = ({ children }) => <>{children}</>;
```

```tsx
import * as React from "react";
import { SFC } from "react";

export const BreadcrumbItem: SFC = ({ children }) => <>{children}</>;
```

```tsx
import React from "react";

export const BreadcrumbItem = ({ children }) => <>{children}</>;
```

### PR DESCRIPTION
**Pull request by malbernaz**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Fixes #7979
| Patch: Bug Fix?          | Yes
| Major: Breaking Change?  | No
| Minor: New Feature?      | No
| Tests Added + Pass?      | Yes
| Documentation PR         | No
| Any Dependency Changes?  | No
| License                  | MIT

This PR Adds `JSXFragment` to plugin-transform-typescript check for type imports.

### CODE REVIEW

**Comment by hzoo** [comment]
Nice, Can you rebase on top of master? the other code changes shouldn't be necessary because of the PR I merged earlier I believe?

**Comment by malbernaz** [comment]
Thanks! Had some trouble to rebase it, but there you go. Think it's ok now

File: packages/babel-plugin-transform-typescript/src/index.js [hunk 1]
```diff
@@ -279,6 +279,9 @@ export default declare((api, { jsxPragma = "React" }) => {
       JSXElement() {
         sourceFileHasJsx = true;
       },
+      JSXFragment() {
+        sourceFileHasJsx = true;
+      },
```
Comment (xtuc) [inline packages/babel-plugin-transform-typescript/src/index.js]: Actually we could just use `JSX() { sourceFileHasJsx = true }` which covers both

Comment (hzoo) [reply packages/babel-plugin-transform-typescript/src/index.js]: Yeah or "JSXElement|JSXFragment(){}", but not sure I care

Comment (xtuc) [reply packages/babel-plugin-transform-typescript/src/index.js]: I don't really care either but I have a preference for the jsx since it's smaller.

**Comment by xtuc** [comment]
That's unexpected, now the tests are failing. The JSX syntax isn't parsed anymore for some reason.

**Comment by malbernaz** [comment]
should I regress?

**Comment by xtuc** [comment]
Well yes but I don't understand why it fails.

**Comment by malbernaz** [comment]
Yes, very strange.. No clue at all of whats happening?

**Comment by xtuc** [comment]
I couldn't reproduce locally so it's hard to debug anyway.

**Comment by malbernaz** [comment]
and it failed again...

**Comment by xtuc** [comment]
I'm looking into it, in previous commit the tests passed.

I also pinged you on Slack, if you want to chat over there.

**Comment by malbernaz** [comment]
ok, thanks

**Comment by xtuc** [comment]
Sorry but i'm not able to reproduce it locally.

**Comment by jquense** [comment]
Just ran across a version of this bug, if you specify the `jsx` pragma in a comment that differs from the Fragment pragma it's still removed.

```js
/** @jsx jsx **/
/** @jsxFrag React.Fragment **/

import { jsx } from '@emotion/core'
import React from 'react'

<><p/></>
```
Output will be:

```js
/** @jsx jsx **/
/** @jsxFrag React.Fragment **/

import { jsx } from '@emotion/core'

jsx(React.Fragment, null, jsx('p))
```

**Pull request merged**